### PR TITLE
Bugfix: Prevents contributors from editing other posts

### LINKED ISSUE #1505: Update Post mutation not properly respecting contributor access control
Currently WPGraphQL allows contributors to update posts created by other users. 

Contributors should only be allowed to edit posts created by themselves.

### PR DESCRIPTION
**Pull request by jasonbahl**
What does this implement/fix? Explain your changes.
---------------------------------------------------
- Adds a check for the UpdatePost mutation to ensure the requesting user has permission to update the post object
- Adds tests for updating others posts with the PostObjectMutation
- Fixes bug with GraphiQL Admin bar showing for users that don't have capabilities to use it

**Before:**

Logged in as a contributor, I can execute a GraphQL `updatePost` mutation and change the title of a post authored by another user. 

Screen Shot 2020-10-13 at 10 04 29 PM

**After:**

Logged in as a contributor, I can execute a GraphQL `updatePost` mutation to change the title of a post authored by another user and get an error in response. 

Does this close any currently open issues?
------------------------------------------
closes #1505

### CODE REVIEW

File: src/Mutation/PostObjectUpdate.php [hunk 1]
```diff
@@ -105,6 +113,10 @@ public static function mutate_and_get_payload( $post_type_object, $mutation_name
 				throw new UserError( sprintf( __( 'Sorry, you are not allowed to update %1$s as this user.', 'wp-graphql' ), $post_type_object->graphql_plural_name ) );
 			}
 
+			/**
```
Comment (smthomas) [inline src/Mutation/PostObjectUpdate.php:116]: It's a bit unclear what this comment is for. Is this just a high level grouping comment for all the code that follows it?

File: src/Mutation/PostObjectUpdate.php [hunk 2]
```diff
@@ -95,6 +95,14 @@ public static function mutate_and_get_payload( $post_type_object, $mutation_name
 				throw new UserError( sprintf( __( 'Sorry, you are not allowed to update a %1$s', 'wp-graphql' ), $post_type_object->graphql_single_name ) );
 			}
 
+			/**
+			 * If the existing post was authored by another author, ensure the requesting user has permission to edit it
+			 */
+			if ( $existing_post->post_author !== get_current_user_id() && ! current_user_can( $post_type_object->cap->edit_others_posts ) ) {
+				// translators: the $post_type_object->graphql_single_name placeholder is the name of the object being mutated
```
Comment (smthomas) [inline src/Mutation/PostObjectUpdate.php:102]: Does this comment get removed at some point or is this just information for translations that will stay in the codebase?

**Review by smthomas — COMMENTED** [review]
Just a few questions on some comments. Otherwise it all looks good to me!

File: src/Mutation/PostObjectUpdate.php (hunk 2 above)
Comment (jasonbahl) [reply src/Mutation/PostObjectUpdate.php:102]: It stays in. Ya, it's used by tooling for translating WordPress plugins so translators know what variables in strings are as they work on translating strings.

File: src/Mutation/PostObjectUpdate.php (hunk 1 above)
Comment (jasonbahl) [reply src/Mutation/PostObjectUpdate.php:116]: 🤔  ya. . .I think I moved the code I was working on up to lines 98-105 but left a comment I had started writing and didn't catch it before committing. . .lol

I'll remove this comment.

File: src/Mutation/PostObjectUpdate.php [hunk 3]
```diff
@@ -105,6 +113,10 @@ public static function mutate_and_get_payload( $post_type_object, $mutation_name
 				throw new UserError( sprintf( __( 'Sorry, you are not allowed to update %1$s as this user.', 'wp-graphql' ), $post_type_object->graphql_plural_name ) );
 			}
 
+			/**
+			 * If the current user is trying to edit a post
+			 */
+
```
Comment (jasonbahl) [inline src/Mutation/PostObjectUpdate.php:119]:
```suggestion
```

**Review by smthomas — APPROVED** [review]

File: tests/wpunit/PostObjectMutationsTest.php [hunk 4]
```diff
@@ -7,6 +7,7 @@ class PostObjectMutationsTest extends \Codeception\TestCase\WPTestCase {
 	public $client_mutation_id;
 	public $admin;
 	public $subscriber;
+	public $contributor;
```
Comment (jasonbahl) [inline tests/wpunit/PostObjectMutationsTest.php:10]:
```suggestion
```

**Pull request merged**